### ISSUE TITLE
Failed to compile: feature `edition 2021` is required

### ISSUE THREAD

**Issue by vedantroy** [bytecodealliance/wasm-tools#457]
I have cargo version: `cargo 1.54.0-nightly (e931e4796 2021-05-24)`.

When trying to install the tools with `cargo install`, I get:

```
error: failed to compile `wasm-tools v1.0.1`, intermediate artifacts can be found at `/tmp/cargo-installbbf1ym`

Caused by:
  failed to parse manifest at `/home/vedantroy/.cargo/registry/src/github.com-1ecc6299db9ec823/wasm-encoder-0.9.0/Cargo.toml`

Caused by:
  feature `edition2021` is required

  consider adding `cargo-features = ["edition2021"]` to the manifest
```

Any ideas what's going on?

**Comment by srenatus** [comment]
FWIW I've been seeing the same thing just now; running `rustup update` resolved this. My cargo version now is:

    cargo 1.58.0 (f01b232bc 2022-01-19)

**Comment by peterhuene** [comment]
I bumped the edition for `wasm-encoder` to 2021, although there's no specific reason that it needed to be.

Although a policy wasn't explicitly stated for this repo, our minimum toolchain version has been "stable" in other BCA repos (e.g. Wasmtime) as it's really not fun to support older toolchains in significantly complex repos as it hampers the adoption of stabilized toolchain features by the contributors.

Like @srenatus, I'm unable to reproduce a problem with `cargo install wasm-tools` using the latest stable tools (cargo at 1.58.0, rustc at 1.58.1).

We can back out the edition change to that crate, if needed, but I do think we should not have a minimum toolchain version other than stable.

**Comment by vedantroy** [comment]
> I'm going to close this since this is intended in the sense that upgrading to 2021 should be fine, but if a toolchain upgrade is too difficult for you @vedantroy we can look into going back to 2018.

Upgrading is fine. I guess I was confused since my cargo/rustc version looks like it was compiled/released/something?? in 2021:

(My cargo version)
```
cargo 1.54.0-nightly (e931e4796 2021-05-24)
```

**Comment by peterhuene** [comment]
The 2021 edition was released with 1.56.0; while the nightly build you were using was indeed released in 2021, it looks like it didn't support the "2021 edition of Rust" at that time.
